grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -534,22 +649,20 @@ private void loadApiConfig(ApiConfig apiConfig) {
 
     private final ManagedChannel delegate;
     private final int channelId;
-    private int affinityCount;
-    private int activeStreamsCount;
+    private final AtomicInteger affinityCount;
+    private final AtomicInteger activeStreamsCount;
```

Comment:
why make them AtomicIntegers. is there a concurrentcy issue on these variables? I am not seeing it.. but maybe I am missing it.
(I see they are accessed in a synchronized method - so that synchronization should take care of the concurrency of access on these in that case. If there is a place in the code where they are accessed concurrently by multiple threads, then yea we need AtomicIntegers. Otherwise it is needless overhead..)

Reply:
These are mutated from GcpClientCall. And yes in one case (https://github.com/nimf/grpc-gcp-java/blob/73362ec89ab1433ab33bcac0ec55109e5ecfd983/grpc-gcp/src/main/java/com/google/grpc/gcp/GcpClientCall.java#L120) it is called from within a synchronized block but that synchronization is only within a particular GcpClientCall and we can have multiple GcpClientCalls running in parallel.
I also experienced flaky tests in this repo when these were not atomic.
Which synchronized method you are referring to? Maybe I am missing something.